ambry-clustermap/src/test/java/com.github.ambry.clustermap/ClusterMapManagerTest.java

```diff
@@ -148,21 +152,67 @@ public void bestEffortAllocation()
     ClusterMapManager clusterMapManager = new ClusterMapManager(partitionLayout);
     List<PartitionId> allocatedPartitions;
 
-    // Allocate a five partitions that fit within cluster's capacity
-    allocatedPartitions = clusterMapManager.allocatePartitions(5, replicaCountPerDataCenter, replicaCapacityInBytes);
+    // Allocate five partitions that fit within cluster's capacity
+    allocatedPartitions =
+        clusterMapManager.allocatePartitions(5, replicaCountPerDataCenter, replicaCapacityInBytes, true, false);
     assertEquals(allocatedPartitions.size(), 5);
     assertEquals(clusterMapManager.getWritablePartitionIds().size(), 5);
 
     // Allocate "too many" partitions (1M) to exhaust capacity. Capacity is not exhausted evenly across nodes so some
     // "free" but unusable capacity may be left after trying to allocate these partitions.
+    allocatedPartitions = clusterMapManager.allocatePartitions(1000 * 1000, replicaCountPerDataCenter,
+        replicaCapacityInBytes, true, false);
+    assertEquals(allocatedPartitions.size() + 5, clusterMapManager.getWritablePartitionIds().size());
+    System.out.println(freeCapacityDump(clusterMapManager, testHardwareLayout.getHardwareLayout()));
+
+    // Capacity is already exhausted...
+    allocatedPartitions =
+        clusterMapManager.allocatePartitions(5, replicaCountPerDataCenter, replicaCapacityInBytes, true, false);
+    assertEquals(allocatedPartitions.size(), 0);
+  }
+
+  @Test
+  public void bestEffortAllocationRackAware()
```

Comment:
ok even this says "bestEffort" but the arg `bestEffort` is always `false`. Perhaps there is no relation b/w the function name and the arg but we should rename the function to avoid confusion maybe?